execution/registry/registry.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2017 Monax Industries Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package registry
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hyperledger/burrow/crypto"
+	"github.com/tendermint/tendermint/p2p"
+)
+
+// NewRegisteredNode takes a p2p node key, and returns a node entry with a signature
+func NewRegisteredNode(nodeKey *p2p.NodeKey, moniker, netAddress string) (*RegisteredNode, []byte, error) {
+	burrowNodePubKey, err := crypto.PublicKeyFromTendermintPubKey(nodeKey.PubKey())
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not convert tm publicKey: %v", err)
+	}
+
+	burrowNodeID, err := crypto.AddressFromHexString(string(nodeKey.ID()))
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not convert tm nodeID: %v", err)
+	}
+
+	node := &RegisteredNode{
+		Moniker:    moniker,
+		NetAddress: netAddress,
+		ID:         burrowNodeID,
+		PublicKey:  burrowNodePubKey,
+	}
+
+	data, err := node.Marshal()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	sig, err := nodeKey.PrivKey.Sign(data)
```

Comment:
Okay now I think I understand the dance we do. We sign using tendermint crypto, then we put public key with our crypto for transport, which we then convert to tendermint the other end.

I think it would be more in line with minimising our explicit dependency on tendermint crypto by signing with _our_ private_key implementation. All we would need for this is a conversion function for private keys like the one for public keys here: https://github.com/hyperledger/burrow/blob/develop/crypto/tendermint.go#L12-L22 (note you cannot just call `Bytes()` because that gives you amino encoded with a prefix, you need to do type switch and case the array).

If we do it this way then we can just use our standard `Verify` function on the public key in the identify tx.

Reply:
As discussed, we're no longer going to transmit the Tendermint public key, just the node ID.